### ISSUE TITLE
Improved way to display matches

### ISSUE THREAD

**Issue by asimonw** [emfoundation/ce100-app#1065]
We'd like to change the way matches are shown on the challenge view.

This would involve the following changes:
+ [x] group the result list in three sections (Organisations, Challenges, Insights)
+ [x] add a title to every section
+ [x] If there are none of a given category, we will not display the category at all.
+ [x] If _nothing_ appears in _any_ categories, the same message that we currently have now appears, encouraging people to tag their challenge better.
+ [x] show the logos for organisations -- to be done once API work is merged in
+ [x] show up to 4 items per section by default
+ [x] add a "See more" link
+ [x] clicking on see more reveals another up to 6 items per section (up to 10 in total). 
+ [x] The "See more" changes to "See less"
+ [x] if less than 4 matches exist for a given section, don't show the "See more" link
+ [x] Repeat for each section

The visual can be found in the Invision prototype.

**Comment by iteles** [comment]
Clicking on 'See more' pushes the other sections underneath it down.

**Comment by iteles** [comment]
If there are none of a given category, we will not display the category at all. If _nothing_ appears in _any_ categories, the same message that we currently have now appears, encouraging people to tag their challenge better.

**Comment by Cleop** [comment]
**Notes for point on displaying the org logo:**
Cannot yet update the challenges plugin to enable us to display org urls because the plugin has been updated by the API work which would break the rest of my branch. So that it's handy and we can remember, this is the change that needs to be made once we can:
![image](https://user-images.githubusercontent.com/16775804/33332734-f5798d66-d45c-11e7-976c-fe9f26b2747b.png)

I've already done the work to display the default logo and once the plugin is updated the normal one should also display.

**Comment by Cleop** [comment]
@fannyBkane - how would you like the see more/less button to look?

**For the time being I've implemented this:**
<img src="https://user-images.githubusercontent.com/16775804/33383967-aa061a8c-d51c-11e7-804c-719220caa31d.png" width=150px />

**Comment by iteles** [comment]
Half a day added to integrate this with API.

**Comment by SimonLab** [comment]
I've now created a PR (#1165 ) which merge most of the matching features with the API features but ow that we get the tags via the API the matching of the orgs query needs to be updated in the plugin.

**Comment by SimonLab** [comment]
The matches are now display based on the tags from the API. However @asimonw noted that the ranking wasn't right on some items: #1109 and I think I spotted a bug linked to this issue yesterday. I'll review #1109 first then I'll put both of this issues in please test once a new version has been deployed

**Comment by SimonLab** [comment]
@asimonw this is now ready to be tested on ce100-test and on ce100-dam-api app which contains a bit more information

**Comment by iteles** [comment]
@Cleop Will need your help to ensure consistency on the hover states across the app - is this something you've already touched on for 'normal' links rather than the nav ones we demo'ed in the last sprint?

**Comment by asimonw** [comment]
@SimonLab Now the orgs in the matches are not actually being pulled in anymore. 
<img width="255" alt="screen shot 2017-12-12 at 23 31 07" src="https://user-images.githubusercontent.com/15235804/33914245-3a3cf620-df95-11e7-9663-703cbd3e603b.png">
It would be great if you could fix that first thing.

**Comment by iteles** [comment]
Adding this as priority 1 mostly for clarity 👍

**Comment by SimonLab** [comment]
@asimonw this should be fixed now, I've deployed the fix on the dam-api application to let you have more data to test with. Let us know if it's ok for you and we can deploy on production just after

**Comment by asimonw** [comment]
@SimonLab That looks better, but there's a lot of vertical space between the orgs in the list. It was less before. Would it be possible to revert that?

**Comment by SimonLab** [comment]
I've fixed the spaces:
![image](https://user-images.githubusercontent.com/6057298/33939132-df8908aa-e001-11e7-8910-9cb34a2bfddf.png)

To be sure you spotted this only on the see more section of the mathing organisations or also somewhere else?
